RecDP/pyrecdp/primitives/operations/rag/document_loader.py

```diff
@@ -0,0 +1,103 @@
+from typing import List, Union, Any, Optional, Dict
+
+from langchain.document_loaders import UnstructuredFileLoader
+from langchain.schema import Document
+
+from pyrecdp.primitives.llmutils.document.reader import (
+    DocumentReader,
+    PDFReader,
+    DocxReader,
+    ImageReader,
+    UnstructuredReader
+)
+from pyrecdp.primitives.operations.base import BaseLLMOperation, LLMOPERATORS
+
+
+class DocumentFileLoader(UnstructuredFileLoader):
+    def __init__(
+            self,
+            file_path: Union[str, List[str]],
+            mode: str = "single",
+            **kwargs: Any):
+        super().__init__(file_path, mode, **kwargs)
+
+        if file_path.endswith(".pdf"):
+            self.loader = PDFReader()
+        elif file_path.endswith(".docx"):
+            self.loader = DocxReader()
+        elif file_path.endswith(".jpeg") or file_path.endswith(".jpg"):
+            self.loader = ImageReader()
+        elif file_path.endswith(".pptx") or file_path.endswith("xlsx"):
+            # use langchain file loader
+            self.loader = UnstructuredReader()
+        else:
+            self.loader = self
+
+    def load(self) -> List[Document]:
+        if isinstance(self.loader, DocumentReader):
+            import pathlib
+            path = pathlib.Path(self.file_path)
+            return [Document(page_content=doc.text, metadata=doc.metadata) for doc in self.loader.load(path)]
+        else:
+            return super().load()
+
+
+class DocumentTextLoader(BaseLLMOperation):
```

Comment:
Based on our discussion in project meeting,  I think @yao531441  proposed we don't wrap langchain, but using a conversion to update text pipeline at [optimize_execute_plan](https://github.com/intel/e2eAIOK/blob/main/RecDP/pyrecdp/LLM/TextPipeline.py#L231), want to have an alignment here